client/components/domains/domain-registration-suggestion/index.jsx

```diff
@@ -185,13 +187,42 @@ class DomainRegistrationSuggestion extends React.Component {
 		);
 	}
 
+	/**
+	 * Splits a domain into name and tld. Everything after the "first dot"
+	 * becomes the tld. This is not very comprehensive since there can be
+	 * subdomains which would fail this test. However, for our purpose of
+	 * highlighting the TLD in domain suggestions, this is good enough.
+	 *
+	 * @param {string} domain The domain to be parsed
+	 */
+	getDomainParts( domain ) {
+		const parts = domain.split( '.' );
+		const name = parts[ 0 ];
+		const tld = `.${ parts.slice( 1 ).join( '.' ) }`;
+		return {
+			name,
+			tld,
+		};
+	}
+
+	renderDomainParts( domain ) {
+		const { name, tld } = this.getDomainParts( domain );
```

Comment:
I wonder if we can use `getUrlParts()`([reference](https://github.com/Automattic/wp-calypso/blob/master/client/lib/url/url-parts.ts#L79)) instead of creating a custom definition?

Reply:
I saw this test for `getUrlParts`:
https://github.com/Automattic/wp-calypso/blob/0d319b7ff8faefff1a876d94d3314221a6a9f01c/client/lib/url/test/url-parts.js#L59-L91

It does not seem to separate the domain into the name and tld (`example` and `.com`).

The closest function I found was [`getTld`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/domains/get-tld.js), but it split `abc.wordpress.com` into [`abc.wordpress`, `.com`].
The expected split is into [`abc`, `.wordpress.com`].

Reply:
Agreed, let's keep `getDomainParts()` then.